### PR TITLE
DNSimple: Implement GetZones and ListZones 

### PR DESCRIPTION
**Pull request by onlyhavecans**
- Implement GetZoneRecords Per #628 
- Update go mod & tidy Per #619 
- Update comments and notes
- Implement ListZones per documentation
- Add a missing error checking
- update `go generate` after updating capabilities

### CODE REVIEW

File: providers/dnsimple/dnsimpleProvider.go [hunk 1]
```diff
@@ -360,6 +377,38 @@ func (c *DnsimpleApi) updateRecordFunc(old *dnsimpleapi.ZoneRecord, rc *models.R
...
+// Returns all the zones in an account
+func (c *DnsimpleApi) ListZones() ([]string, error) {
+	client := c.getClient()
+	accountID, err := c.getAccountID()
+	if err != nil {
+		return nil, err
+	}
+
+	opts := &dnsimpleapi.ZoneListOptions{}
+	zs := []dnsimpleapi.Zone{}
```
Comment (tlimoncelli) [inline providers/dnsimple/dnsimpleProvider.go]: zs is used to collect `dnsimpleapi.Zone`'s which is used to later collect just the `Name` field from each.  Suggestion: just collect the `Name` field.

i.e. This line would become: `zones := []string` and then line 396 would become a loop that collects the zone names.  The loop around line 406 would be unneeded.

File: docs/writing-providers.md [hunk 2]
```diff
@@ -92,14 +92,14 @@ Pick a similar provider as your base.  Providers basically fall
 into three general categories:
 
 * **zone:** The API requires you to upload the entire zone every time. (BIND).
-* **incremental-record:** The API lets you add/change/delete individual DNS records. (ACTIVEDIR, CLOUDFLARE, NAMEDOTCOM, GCLOUD, ROUTE53)
+* **incremental-record:** The API lets you add/change/delete individual DNS records. (ACTIVEDIR, CLOUDFLARE, DNSIMPLE, NAMEDOTCOM, GCLOUD, ROUTE53)
```
Comment (tlimoncelli) [inline docs/writing-providers.md]: Thank you for updating the docs!

File: providers/dnsimple/dnsimpleProvider.go (hunk 1 above)
Comment (onlyhavecans) [reply providers/dnsimple/dnsimpleProvider.go]: Thanks for the feedback! Updated 🎊

**Comment by tlimoncelli** [comment]
Thanks for the PR!!

**Pull request merged**